docs/components/configMaps.md

```diff
@@ -0,0 +1,146 @@
+maps to other components. However, the grade can be extended to provide custom configuration options.
+
+## Model Structure
+
+The grade stores the model data in the following format:
+
+``` javascript
+model: {
+    map: {
+        buttons: {
+            "0": {
+                defaultAction: "previousPageInHistory",
+                currentAction: null,
+                speedFactor: 1,
+                background: false
+            },
+            // The buttons below use same format as above but have different values for configuration options (keys).
+            "1": {...},
+            "2": {...},
+            "3": {...},
+            "4": {...},
+            "5": {...},
+            "6": {...},
+            "7": {...},
+            "8": {...},
+            "9": {...},
+            "10": {...},
+            "11": {...},
+            "12": {...},
+            "13": {...},
+            "14": {...},
+            "15": {...},
+            // The buttons below are reserved for special functionality so they cannot be configured.
+            "16": null,
+            "17": null
+        },
+        axes: {
+            "0": {
+                defaultAction: "scrollHorizontally",
+                currentAction: null,
+                speedFactor: 1,
+                invert: false
+            },
+            // The axes below use the same format as above but have different values for configuration options (keys).
+            "1": {...},
+            "2": {...},
+            "3": {...}
+        }
+    },
+    // Stores all the common configuration options for the gamepad actions.
+    commonConfiguration: {
+        homepageURL: "https://www.google.com/"
+    }
+}
+```
+
+The first 16 buttons and all the axes are configurable. The configuration options used in their model data are explained
+below:
+
+| Option | Supported Input | Description |
+| :---: | :---: | :--- |
+| `defaultAction` | `buttons` and `axes` | The default action triggered with the particular gamepad input if the `currentAction` is not provided. |
+| `currentAction` | `buttons` and `axes` | The action triggered with the particular gamepad input. It takes precedence over the `defaultAction` and is `null` by default. However, this can be specified when the component is created. |
+| `speedFactor` | `buttons` and `axes` | The speed with which the given action should be executed. It is only applicable if the action is continuous and hence supports only a few actions. |
+| `background` | `buttons` | Specifies whether the new tab or window should be opened in background and is applicable for those two actions only. |
+| `invert` | `axes` | Supported only for actions provided to `axes` and specifies whether the direction of the action should be inverted. |
+
+The option `homepageURL` inside `commonConfiguration` option specifies the URL of the webpage that should be loaded when
```

Comment:
I missed this on the first reading, it's important enough to highlight better.  I'd suggest making a table in the component options" section.  The first entry would be `model.commonConfiguration.homepageURL`.   The second would be `model.map`, summarising briefly what the map does and saying "see below for more details".